### PR TITLE
Firmware catalog resource

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- Create Firmware Catalogs with types "DELL_ONLINE" "NFS" "CIFS" "HTTP" and "HTTPS"

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Firmware Catalog Resource

###### Code Coverage
image

###### Test Output
```
$ MOCKEY_CHECK_GCFLAGS=false go test -timeout 1h -v -run TestFirmwareCatalogResource
=== RUN   TestFirmwareCatalogResourceCreate
--- PASS: TestFirmwareCatalogResourceCreate (12.09s)
=== RUN   TestFirmwareCatalogResourceValidationCreateError
--- PASS: TestFirmwareCatalogResourceValidationCreateError (9.83s)
=== RUN   TestFirmwareCatalogResourceReadCreateUpdateErrors
--- PASS: TestFirmwareCatalogResourceReadCreateUpdateErrors (12.65s)
PASS
ok      terraform-provider-ome/ome      35.828s

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ List the GitHub issues impacted by this PR:
 - [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
 - [ ] I have verified that new and existing unit tests pass locally with my changes
 - [ ] I have not allowed coverage numbers to degenerate
-- [ ] I have maintained at least 90% code coverage
+- [ ] I have maintained at least 80% code coverage
```
Comment (doriac11) [inline .github/PULL_REQUEST_TEMPLATE.md:39]: Updated this to be consistent with the other providers

**Review by rounak-adhikary — APPROVED** [review]
The regex validations are very cool!

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by shekhar-j — APPROVED** [review]

File: ome/resource_firmware_catalog.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+		resp.Diagnostics.AddError(
+			`Unable to read specific firmware catalog: `+currentState.Name.ValueString()+``, err.Error(),
+		)
+	}
+
+	// Set the tf state after read
+	state, mapErr := helper.SetStateCatalogFirmware(ctx, cat, currentState)
+	if mapErr != nil {
+		resp.Diagnostics.AddError(
+			`Unable to process catalog after create: `+state.Name.ValueString()+`.`, mapErr.Error(),
```
Comment (nikitajoshi1) [inline ome/resource_firmware_catalog.go:163]: Update Error message here.

File: ome/resource_firmware_catalog.go [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+			`Unable to update catalog: `+state.Name.ValueString()+``, err.Error(),
+		)
+		return
+	}
+
+	// Update tf state after update of catalog
+	finalState, mapErr := helper.SetStateCatalogFirmware(ctx, cat, plan)
+	if mapErr != nil {
+		resp.Diagnostics.AddError(
+			`Unable to process catalog after create: `+state.Name.ValueString()+`.`, mapErr.Error(),
```
Comment (nikitajoshi1) [inline ome/resource_firmware_catalog.go:267]: Error message should be changed.

**Review by nikitajoshi1 — DISMISSED** [review]

File: ome/resource_firmware_catalog.go [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+	tflog.Trace(ctx, "firmwareCatalogResource: update end")
+}
+
+// ImportState implements resource.Resource.
+func (r *firmwareCatalogResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	var importState models.OmeSingleCatalogResource
+	tflog.Trace(ctx, "firmwareCatalogResource: import state started")
+	// Create Session and defer the remove session
+	omeClient, d := r.p.createOMESession(ctx, "resource_discovery Read")
```
Comment (nikitajoshi1) [inline ome/resource_firmware_catalog.go:282]: resource_discovery -> resource_firmwarwe_catalog

File: ome/resource_firmware_catalog.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+	if valError != nil {
+		resp.Diagnostics.AddError(
+			`Unable to update catalog, validation error: `, valError.Error(),
+		)
+		return
+	}
+
+	// Create Session and defer the remove session
+	omeClient, d := r.p.createOMESession(ctx, "resource_discovery Create")
```
Comment (nikitajoshi1) [inline ome/resource_firmware_catalog.go:239]: resource_discovery -> resource_firmwarwe_catalog (For all methods)

**Review by shekhar-j — APPROVED** [review]

**Review by nikitajoshi1 — APPROVED** [review]

**Pull request merged**